workers/src/executor_def.rs

```diff
@@ -0,0 +1,152 @@
+                           Self::handle_verified_packet(&contract, &mut self.packet_queue, inner_log.data()).await?;
+                        },
+                        Err(e) => { error!("Failed to decode PacketVerified event: {:?}", e);}
+                    }
+                },
+            }
+            if self.finish {
+                break;
+            }
+        }
+        Ok(())
+    }
+
+    async fn handle_verified_packet(
+        contract: &ContractInst,
+        queue: &mut VecDeque<PacketSent>,
+        packet_verified: &PacketVerified,
+    ) -> Result<()> {
+        if queue.is_empty() {
+            return Ok(());
+        }
+
+        let packet_sent = queue.pop_front().unwrap();
+        // We don't expect any more items to be present. If we have any - they are garbage/leftovers.
+        queue.clear();
+        // Despite being described with other arguments, the only real implementation of
+        //  `executable` function is in the contract located here: https://shorturl.at/4H6Yz
+        // function executable(Origin memory _origin, address _receiver) returns (ExecutionState)
+        let call_builder = contract.function(
+            "executable",
+            &[
+                prepare_header(&packet_sent.encodedPayload[..]),
+                DynSolValue::Address(packet_verified.receiver),
+            ],
+        )?;
+
+        let mut retry_count = 0;
+        // status `Executable` is represented by the integer 2 in the enum.
+        // To read more: https://tinyurl.com/zur3btzs (line 9)
+        let executable: &DynSolValue = &DynSolValue::Int(I256::unchecked_from(2), 32);
+        loop {
+            if retry_count == Self::MAX_RETRIES {
+                error!("Maximum retries reached while waiting for `Executable` state.");
+                break;
+            }
+            let call_result = call_builder.call().await?;
+            if call_result.len() != 1 {
+                error!("`executable` function call returned invalid response.");
+                break;
+            }
+
+            // Note: why not pattern matching here? Rust analyzer ranted on `executable` variable
+            // in the pattern, so an author decided to make it via conditions.
+            if call_result[0].eq(NFFLExecutor::NOT_EXECUTABLE)
+                || call_result[0].eq(NFFLExecutor::VERIFIED_NOT_EXECUTABLE)
+            {
+                // state: NotExecutable or VerifiedNotExecutable, await commits/verifications
+                sleep(Duration::from_secs(1)).await;
+                retry_count += 1;
+                continue;
+            } else if call_result[0].eq(executable) {
+                // state: Executable, fire and forget `lz_receive`
+                lz_receive(contract, &packet_sent.encodedPayload[..]).await?;
+                break;
+            } else {
+                // We ignore Executed status, it just frees the executor.
+                break;
+            }
```

Comment:
```suggestion
            match &call_result[0] {
                x if x.eq(NFFLExecutor::NOT_EXECUTABLE) || x.eq(NFFLExecutor::VERIFIED_NOT_EXECUTABLE) => {
                    // state: NotExecutable or VerifiedNotExecutable, await commits/verifications
                    sleep(Duration::from_secs(1)).await;
                    retry_count += 1;
                    continue;
                }
                x if x.eq(executable) => {
                    // state: Executable, fire and forget `lz_receive`
                    lz_receive(contract, &packet_sent.encodedPayload[..]).await?;
                    break;
                }
                _ => {
                    // We ignore Executed status, it just frees the executor.
                    break;
                }
            }
```
eh this is just style but you can ignore it. nitpick

Reply:
Em. Sorry, mate, your variant is less preferable to me.